tobias/utils/motifs.py

```diff
@@ -477,6 +531,93 @@ def find_best_pair(cluster_motifs, score_dict):
 
 	return best_similarity_motifs
 
+#--------------------------------------------------------------------------------------------------------#
+def get_formation(formation, ncol, nrow, nmotifs):
+	""" check formation or set formation to one of the existing options """
+
+	# if ncol and/or nrow is missing automatically set fitting parameters 
+	if formation != "alltoone":
+		if ncol is None and nrow is None:
+			half_nmotifs = math.ceil(math.sqrt(nmotifs))
+			ncol, nrow = half_nmotifs, half_nmotifs
+		else:
+			if ncol is None:
+				ncol = math.ceil(nmotifs/nrow)
+			if nrow is None:
+				nrow = math.ceil(nmotifs/ncol)
+
+	if isinstance(formation, str):
+		
+		if formation == "row":
+
+			# fill plot left to right
+
+			formation = list()
+			rows = list(range(nrow))
+			for row in rows:
+				for col in range(ncol):
+					formation.append((row,col))
+
+		elif formation == "col":
+
+			# fill plot top to bottom
+
+			formation = list()
+			rows = list(range(nrow))
+			for col in range(ncol):
+				for row in rows:
+					formation.append((row,col))
+
+		elif formation == "alltoone":
+
+			# fill first column execpt for one motif
+			# ignores parameter ncol and nrow
+
+			formation = list()
+			rows = list(range(nmotifs-1))
+			for row in rows:
+				formation.append((row,0))
+			formation.append((math.ceil(len(rows)/2)-1, 1))
+
+			ncol = 2
+			nrow = len(rows)
+
+		else:
+			sys.exit("ERROR: Unknown formation setting.")
+	else:
+
+		# Check if formation fits to grid
+		formation_max_row = max([i[0] for i in formation])
+		formation_max_col = max([i[1] for i in formation])
+		if nrow < formation_max_row or ncol < formation_max_col:
+			sys.exit("ERROR: Grid is to small for specified formation")
+
+	return formation, nrow, ncol
+
+#--------------------------------------------------------------------------------------------------------#
+def create_logo(counts, ax, max_x):

```

Comment:
What about including create_logo in OneMotif? Then you can pull the counts directly from the object and  MotifList.plot_motifs can immediately use the object to plot to axes.

Reply:
Also; what is max_x? The length of the motif? Would be nice if this was chosen dynamically.

Reply:
I need the max_x (I have to rename the parameter!) for plotting the Motif list. It contains the length of the longest motif in the list. Without it, I would not be able to align the axis and I am not aware of another solution to this problem. I could make this parameter optional and set it to df.shape[0] as default.

![motif_plot](https://user-images.githubusercontent.com/36538452/72902462-c8cd3080-3d2b-11ea-8ead-30761f93d6b4.png)
![motif_plot_2](https://user-images.githubusercontent.com/36538452/72902467-ca96f400-3d2b-11ea-982f-168840f6b2d1.png)

These images show the difference. It depends on what you prefer.